pkg/controller/certificaterequests/test/util.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2019 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/pem"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+	testpkg "github.com/jetstack/cert-manager/pkg/controller/test"
+	"github.com/jetstack/cert-manager/pkg/issuer"
+	"github.com/jetstack/cert-manager/pkg/util/pki"
+)
+
+func GenerateRSAPrivateKey(t *testing.T) *rsa.PrivateKey {
+	pk, err := pki.GenerateRSAPrivateKey(2048)
+	if err != nil {
+		t.Errorf("failed to generate private key: %v", err)
+		t.FailNow()
+	}
+	return pk
+}
+
+func GenerateCSR(t *testing.T, secretKey crypto.Signer) []byte {
```

Comment:
`t` is not used here and this function smells a lot like functions in `pkg/util/pki` - can the callsites be refactored to pass in something that allows greater configuration? We are statically setting the `CommonName` to `test` here, which is awkward behaviour to integrate in other packages

Reply:
```
	if err != nil {
		t.Error(err)
		t.FailNow()
	}
```

```t``` does get used here. How would you suggest the function look/be moved to?